// @draftbit/core/lib/module/components/RadioButton/RadioButton.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import * as React from "react";
import Config from "../Config";
import IconButton from "../IconButton";
import { getValueForRadioButton } from "../../utilities";
import { useRadioButtonGroupContext } from "./context";

const RadioButton = _ref => {
  let {
    Icon,
    disabled = false,
    color,
    value = "",
    selected,
    unselectedColor,
    onPress,
    size = Config.radioButtonSize,
    selectedIcon = "MaterialIcons/radio-button-checked",
    unselectedIcon = "MaterialIcons/radio-button-unchecked",
    style,
    ...rest
  } = _ref;
  const {
    value: contextValue,
    onValueChange
  } = useRadioButtonGroupContext();
  const realValue = getValueForRadioButton(value);
  const realContextValue = getValueForRadioButton(contextValue);
  const isSelected = selected !== null && selected !== void 0 ? selected : realContextValue === realValue;

  const handlePress = () => {
    onPress === null || onPress === void 0 ? void 0 : onPress(realValue);
    onValueChange === null || onValueChange === void 0 ? void 0 : onValueChange(realValue);
  };

  return /*#__PURE__*/React.createElement(IconButton, _extends({
    Icon: Icon,
    icon: isSelected ? selectedIcon : unselectedIcon,
    color: isSelected ? color : unselectedColor,
    disabled: disabled,
    onPress: handlePress,
    size: size,
    style: style
  }, rest));
};

export default RadioButton;
