// @draftbit/core/lib/commonjs/components/Swiper/Swiper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _reactNativeWebSwiper = _interopRequireDefault(require("react-native-web-swiper"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Swiper = _ref => {
  let {
    vertical = false,
    loop = false,
    timeout = 0,
    from = 0,
    prevTitle = "",
    nextTitle = "",
    prevTitleColor,
    nextTitleColor,
    dotsTouchable = true,
    dotColor,
    dotActiveColor,
    data,
    keyExtractor,
    renderItem,
    children,
    style
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: style
  }, /*#__PURE__*/_react.default.createElement(_reactNativeWebSwiper.default, {
    from: from,
    loop: loop,
    timeout: timeout,
    vertical: vertical,
    controlsProps: {
      prevTitle,
      nextTitle,
      prevTitleStyle: {
        color: prevTitleColor
      },
      nextTitleStyle: {
        color: nextTitleColor
      },
      dotsTouchable,
      ...(dotColor ? {
        dotProps: {
          badgeStyle: {
            backgroundColor: dotColor
          }
        }
      } : {}),
      ...(dotActiveColor ? {
        dotActiveStyle: {
          backgroundColor: dotActiveColor
        }
      } : {})
    }
  }, data && renderItem ? data.map((item, index) => {
    const component = renderItem({
      item,
      index
    });

    if (!component) {
      return null;
    }

    const key = keyExtractor ? keyExtractor(item, index) : index;
    return /*#__PURE__*/_react.default.cloneElement(component, {
      key
    });
  }) : children));
};

var _default = Swiper;
exports.default = _default;
