// @draftbit/core/lib/commonjs/mappings/Accordion.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;

var _types = require("@draftbit/types");

const SEED_DATA = {
  name: "Accordion",
  tag: "AccordionGroup",
  description: "An expandable container containing components",
  category: _types.COMPONENT_TYPES.container,
  layout: {
    paddingTop: 8,
    paddingRight: 8,
    paddingBottom: 8,
    paddingLeft: 8,
    fontSize: 16
  },
  props: {
    label: (0, _types.createTextProp)({
      label: "Label"
    }),
    expanded: (0, _types.createStaticBoolProp)({
      label: "Expanded",
      description: "Whether the AccordionGroup should be initaially expanded or not"
    }),
    openColor: (0, _types.createColorProp)({
      label: "Open text color"
    }),
    closedColor: (0, _types.createColorProp)({
      label: "Closed text color"
    }),
    caretColor: (0, _types.createColorProp)({
      label: "Caret color"
    }),
    caretSize: (0, _types.createNumberProp)({
      label: "Caret size",
      defaultValue: 24
    }),
    icon: (0, _types.createIconProp)(),
    iconSize: (0, _types.createNumberProp)({
      label: "Icon size",
      defaultValue: 24
    })
  }
};
exports.SEED_DATA = SEED_DATA;
