// @draftbit/core/lib/commonjs/mappings/Banner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;

var _types = require("@draftbit/types");

const SEED_DATA = {
  name: "Banner",
  tag: "Banner",
  category: _types.COMPONENT_TYPES.button,
  props: {
    icon: (0, _types.createIconProp)({
      defaultValue: null,
      required: false
    }),
    content: (0, _types.createTextProp)({
      label: "Content",
      description: "Banner text content"
    }),
    initiallyVisible: (0, _types.createBoolProp)({
      group: _types.GROUPS.basic,
      label: "Initially visible",
      description: "Whether the banner should be visible",
      defaultValue: true
    }),
    dismissable: (0, _types.createBoolProp)({
      group: _types.GROUPS.basic,
      label: "Dismissable",
      description: "Whether the banner should be able to be closed",
      defaultValue: true
    }),
    buttonColor: (0, _types.createColorProp)({
      label: "Button color",
      defaultValue: "primary"
    })
  }
};
exports.SEED_DATA = SEED_DATA;
