// @draftbit/core/lib/commonjs/mappings/CheckboxRow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;

var _types = require("@draftbit/types");

const SEED_DATA = {
  name: "Checkbox Row",
  tag: "CheckboxRow",
  category: _types.COMPONENT_TYPES.input,
  stylesPanelSections: [_types.StylesPanelSections.Typography, _types.StylesPanelSections.Margins, _types.StylesPanelSections.Effects],
  layout: {
    minHeight: 50
  },
  triggers: [_types.Triggers.OnPress, _types.Triggers.OnCheck, _types.Triggers.OnUncheck],
  props: {
    fieldName: (0, _types.createFieldNameProp)({
      defaultValue: "checkboxRowValue",
      valuePropName: "value",
      handlerPropName: "onPress"
    }),
    label: (0, _types.createTextProp)({
      label: "Label",
      description: "Label to show with the checkbox",
      required: true,
      defaultValue: "First Option"
    }),
    labelStyle: (0, _types.createTextStyle)({
      label: "Label Style",
      description: "Change the styles of the label",
      required: false,
      editable: false
    }),
    direction: (0, _types.createRowDirectionProp)(),
    color: (0, _types.createColorProp)({
      description: "Color for the button (used when the checkbox is checked)"
    }),
    uncheckedColor: (0, _types.createColorProp)({
      label: "Unselected Color",
      description: "Color for the button when the checkbox is unchecked"
    }),
    disabled: (0, _types.createBoolProp)({
      label: "Disabled",
      description: "Whether the checkbox is disabled"
    }),
    size: (0, _types.createStaticNumberProp)({
      label: "Size",
      description: "Specifies the size of the icon",
      defaultValue: null
    }),
    checkedIcon: (0, _types.createIconProp)({
      label: "Checked Icon",
      description: 'Icon to show when the checkbox status is "checked"',
      defaultValue: null
    }),
    uncheckedIcon: (0, _types.createIconProp)({
      label: "Unchecked Icon",
      description: 'Icon to show when the checkbox status is "unchecked"',
      defaultValue: null
    })
  }
};
exports.SEED_DATA = SEED_DATA;
