// @draftbit/core/lib/commonjs/mappings/LinearGradient.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;

var _types = require("@draftbit/types");

const SEED_DATA = {
  name: "Linear Gradient",
  tag: "LinearGradient",
  description: "Linear Gradient Component",
  layout: {
    width: "100%",
    height: "100%"
  },
  props: {
    endY: {
      group: _types.GROUPS.basic,
      label: "End Y",
      description: "End position from Bottom",
      editable: true,
      required: false,
      defaultValue: 100,
      min: 0,
      max: 100,
      step: 1,
      precision: 0,
      formType: _types.FORM_TYPES.number,
      propType: _types.PROP_TYPES.NUMBER
    },
    endX: {
      group: _types.GROUPS.basic,
      label: "End X",
      description: "End position from Right",
      editable: true,
      required: false,
      defaultValue: 100,
      min: 0,
      max: 100,
      step: 1,
      precision: 0,
      formType: _types.FORM_TYPES.number,
      propType: _types.PROP_TYPES.NUMBER
    },
    startY: {
      group: _types.GROUPS.basic,
      label: "Start Y",
      description: "Start position from Top",
      editable: true,
      required: false,
      defaultValue: 0,
      min: 0,
      max: 100,
      step: 1,
      precision: 0,
      formType: _types.FORM_TYPES.number,
      propType: _types.PROP_TYPES.NUMBER
    },
    startX: {
      group: _types.GROUPS.basic,
      label: "Start X",
      description: "Start position from Left",
      editable: true,
      required: false,
      defaultValue: 0,
      min: 0,
      max: 100,
      step: 1,
      precision: 0,
      formType: _types.FORM_TYPES.number,
      propType: _types.PROP_TYPES.NUMBER
    },
    color3: (0, _types.createColorProp)({
      label: "Color 3",
      defaultValue: undefined
    }),
    color2: (0, _types.createColorProp)({
      label: "Color 2",
      defaultValue: "secondary"
    }),
    color1: (0, _types.createColorProp)({
      label: "Color 1",
      defaultValue: "primary"
    })
  }
};
exports.SEED_DATA = SEED_DATA;
