// @draftbit/core/lib/commonjs/mappings/WebView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SEED_DATA = void 0;

var _types = require("@draftbit/types");

const SEED_DATA = {
  name: "Web View",
  tag: "WebView",
  description: "Render web content inside a view",
  category: _types.COMPONENT_TYPES.media,
  stylesPanelSections: [_types.StylesPanelSections.Size, _types.StylesPanelSections.Margins, _types.StylesPanelSections.Borders],
  layout: {
    flex: 1
  },
  props: {
    source: (0, _types.createSourceProp)({
      defaultValue: "https://reactnative.dev"
    }),
    optimizeVideoChat: (0, _types.createStaticBoolProp)({
      defaultValue: false,
      label: "Optimize Video Chat",
      description: "Allows for a better experience from web hosted video chat services"
    }),
    javaScriptEnabled: (0, _types.createStaticBoolProp)({
      defaultValue: true,
      group: _types.GROUPS.advanced,
      label: "Enable Javascript",
      description: "Boolean value to enable JavaScript in the WebView. The default value is true."
    }),
    javaScriptCanOpenWindowsAutomatically: (0, _types.createStaticBoolProp)({
      defaultValue: false,
      group: _types.GROUPS.advanced,
      label: "Allow New Windows",
      description: "A Boolean value indicating whether JavaScript can open windows without user interaction. The default value is false."
    }),
    showsHorizontalScrollIndicator: (0, _types.createStaticBoolProp)({
      defaultValue: true,
      group: _types.GROUPS.advanced,
      label: "Horizontal Scollbars",
      description: "Boolean value that determines whether a horizontal scroll indicator is shown in the WebView. The default value is true."
    }),
    showsVerticalScrollIndicator: (0, _types.createStaticBoolProp)({
      defaultValue: true,
      group: _types.GROUPS.advanced,
      label: "Vertical Scollbars",
      description: "Boolean value that determines whether a vertical scroll indicator is shown in the WebView. The default value is true."
    }),
    mediaPlaybackRequiresUserAction: (0, _types.createStaticBoolProp)({
      label: "Media Playback Requires User Action",
      group: _types.GROUPS.advanced,
      description: "Boolean that determines whether HTML5 audio and video requires the user to tap them before they start playing. The default value is true. (Android API minimum version 17)."
    }),
    startInLoadingState: (0, _types.createStaticBoolProp)({
      label: "Start In Loading State",
      group: _types.GROUPS.advanced,
      description: "Boolean value that forces the WebView to show the loading view on the first load. This prop must be set to true in order for the renderLoading prop to work."
    }),
    allowFileAccessFromFileURLs: (0, _types.createStaticBoolProp)({
      label: "Allow File Access From File URLs",
      group: _types.GROUPS.advanced,
      description: "Boolean that sets whether JavaScript running in the context of a file scheme URL should be allowed to access content from other file scheme URLs. The default value is false."
    }),
    allowUniversalAccessFromFileURLs: (0, _types.createStaticBoolProp)({
      label: "Allow Universal Access From File URLs",
      group: _types.GROUPS.advanced,
      description: "Boolean that sets whether JavaScript running in the context of a file scheme URL should be allowed to access content from any origin. Including accessing content from other file scheme URLs. The default value is false."
    }),
    cacheEnabled: (0, _types.createStaticBoolProp)({
      label: "Enable Cache",
      defaultValue: true,
      description: "Sets whether WebView should use browser caching."
    }),
    incognito: (0, _types.createStaticBoolProp)({
      label: "Use Incognito Mode",
      description: "Does not store any data within the lifetime of the WebView."
    }),
    userAgent: (0, _types.createTextProp)({
      group: _types.GROUPS.advanced,
      defaultValue: null,
      label: "User Agent",
      description: "Sets the user-agent for the WebView."
    }),
    applicationNameForUserAgent: (0, _types.createTextProp)({
      group: _types.GROUPS.advanced,
      defaultValue: null,
      label: "User Agent App Name",
      description: "Append to the existing user-agent. Setting userAgent will override this."
    })
  }
};
exports.SEED_DATA = SEED_DATA;
