// @draftbit/core/lib/commonjs/components/DatePicker/DatePickerComponent.web.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _dateFns = _interopRequireDefault(require("@date-io/date-fns"));

var _pickers = require("@material-ui/pickers");

var _styles = require("@material-ui/core/styles");

var _theming = require("../../theming");

var _DefaultTheme = _interopRequireDefault(require("../../styles/DefaultTheme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const DatePickerComponent = _ref => {
  let {
    value,
    onChange,
    mode,
    toggleVisibility,
    isVisible,
    theme
  } = _ref;
  const internalTheme = (0, _styles.createMuiTheme)({
    palette: {
      primary: {
        main: theme.colors.primary || _DefaultTheme.default.colors.primary
      },
      secondary: {
        main: theme.colors.secondary || _DefaultTheme.default.colors.secondary
      }
    }
  });
  const Picker = mode === "date" ? _pickers.DatePicker : mode === "time" ? _pickers.TimePicker : _pickers.DateTimePicker;
  return /*#__PURE__*/_react.default.createElement(_pickers.MuiPickersUtilsProvider, {
    utils: _dateFns.default
  }, /*#__PURE__*/_react.default.createElement(_styles.ThemeProvider, {
    theme: internalTheme
  }, /*#__PURE__*/_react.default.createElement(Picker, {
    value: value,
    open: isVisible,
    onChange: d => {
      toggleVisibility();
      onChange(null, d);
    },
    onClose: () => toggleVisibility(),
    variant: "dialog",
    TextFieldComponent: () => null
  })));
};

var _default = (0, _theming.withTheme)(DatePickerComponent);

exports.default = _default;
