// @draftbit/core/lib/module/components/DatePicker/DatePickerComponent.web.js
import React from "react";
import DateFnsUtils from "@date-io/date-fns";
import { MuiPickersUtilsProvider, DatePicker, TimePicker, DateTimePicker } from "@material-ui/pickers";
import { createMuiTheme, ThemeProvider } from "@material-ui/core/styles";
import { withTheme } from "../../theming";
import Theme from "../../styles/DefaultTheme";

const DatePickerComponent = _ref => {
  let {
    value,
    onChange,
    mode,
    toggleVisibility,
    isVisible,
    theme
  } = _ref;
  const internalTheme = createMuiTheme({
    palette: {
      primary: {
        main: theme.colors.primary || Theme.colors.primary
      },
      secondary: {
        main: theme.colors.secondary || Theme.colors.secondary
      }
    }
  });
  const Picker = mode === "date" ? DatePicker : mode === "time" ? TimePicker : DateTimePicker;
  return /*#__PURE__*/React.createElement(MuiPickersUtilsProvider, {
    utils: DateFnsUtils
  }, /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: internalTheme
  }, /*#__PURE__*/React.createElement(Picker, {
    value: value,
    open: isVisible,
    onChange: d => {
      toggleVisibility();
      onChange(null, d);
    },
    onClose: () => toggleVisibility(),
    variant: "dialog",
    TextFieldComponent: () => null
  })));
};

export default withTheme(DatePickerComponent);
